File: Python/1/forLoop.py
"""
for var in iterable:
    statement
    statement
    ...
else:
    statement
    ...

range(start, stop, step)
start: giá trị ban đầu, mặc định = 0
stop: giá trị cuối cùng
step: bước nhảy của dãy
"""
#Ex1
a = range(1, 10, 1)
for i in a:
    print(i, end = " ")

print("\n---")
#Ex2
for i in range(1, 10, 2):
    print(i, end = " ")

print("\n---")
#Ex3
for i in range(10):
    print(i, end = " ")

print("\n---")
#Ex4
for i in range(20):
    print(i, "lmao")

print("\n---")
#tính tổng S(n) = 1 + 2 + 3 +...+ n
n = int(input())
s = 0
for i in range(1, n + 1):
    s += i
print(s)

print("\n---")
#tính tổng S(n) = 1^2 + 2^2 + 3^2 +...+ n^2
s = 0
for i in range(1, n + 1):
    s += i**2
print(s)

print("\n---")
#tính tích các giai thừa A(n) = 1!.2!.3!...n!
s = 1
for i in range(1, n + 1):
    s *= i
print(s)

print("\n---")
#thay đổi biến i trong vòng for ko làm thay đổi biến i trả về trong vòng for, biến i trong vòng for chỉ phụ thuộc vào hàm range()
for i in range(1, 9):
    print(i)
    i += 2

print("\n---")
#forLoop với else
for i in range(1, 9):
    print(i)
else:
    print("END")

print("\n---")
#forLoop với break
for i in range(1, 21):
    print(i, end = ' ')
    if i % 7 == 0:
        break
    print('Lmao')

print("\n---")
#forLoop với continue
for i in range(1, 9):
    if i == 3:
        continue
    print(i)

print("\n---")
#nested loop
for i in range(3):
    for j in range(2):
        print(i, j)

for i in range(1, 10, 0):
    print(i)

File: Java/BasicJava/src/ForLoop.java
public class ForLoop {
    public static void main(String[] args){
        for(int i = 1; i <= 3 ; i++){
            System.out.println(i);
        }

        System.out.println("---");
        for(int i = 1; i == 3 ; i++){
            System.out.println(i);
        }
        System.out.println("lmao");

        //Tính tổng từ 1 đến n
        System.out.println("---");
        int n = 9;
        long sum = 0;
        for(int i = 0; i <= n; i++){
            sum += n;
        }
        System.out.println(sum);
        
        //Nested loop
        System.out.println("---");
        for(int i = 1; i <= 3; i++){
            System.out.println(i);
            for(int j = 1; j <= 2; j++){
                System.out.println(j);
            }
            System.out.println("lmao");
        }

        //break-continue
        System.out.println("---");
        for(int i = 0; i <= 9; i++){
            if(i == 5){
                break;
            }
            System.out.println(i);
        }
        
        //
        System.out.println("---");
        outer: for(int i = 0; i <= 3; i++){
            for(int j = 0; j <= 2; j++){
                if(i == 1){
                    break outer;
                }
                System.out.println(j);
            }
            System.out.println(i);
        }

        //
        System.out.println("---");
        for(int i = 1; i <= 5; i++){
            if(i == 3){
                continue;
            }
            System.out.println(i);

        }


    }
}
